crates/sui-transaction-builder/src/lib.rs

```diff
@@ -674,6 +674,36 @@ impl TransactionBuilder {
         ))
     }
 
+    pub async fn finish_programmable_transaction(
+        &self,
+        signer: SuiAddress,
+        builder: ProgrammableTransactionBuilder,
+        gas: Option<ObjectID>,
+        gas_budget: u64,
+    ) -> anyhow::Result<TransactionData> {
+        let pt = builder.finish();
+        let all_inputs = pt.input_objects()?;
+        let inputs = all_inputs
+            .iter()
+            .flat_map(|obj| match obj {
+                InputObjectKind::ImmOrOwnedMoveObject((id, _, _)) => Some(*id),
+                _ => None,
+            })
+            .collect();
+        let gas_price = self.0.get_reference_gas_price().await?;
+        let gas = self
+            .select_gas(signer, gas, gas_budget, inputs, gas_price)
+            .await?;
+
+        Ok(TransactionData::new(
+            TransactionKind::programmable(pt),
+            signer,
+            gas,
+            gas_budget,
+            gas_price,
+        ))
+    }
+
```

Comment:
Should this be in `light-client-helpers`?

Reply:
I think it is better to have it inside the `TransactionBuilder` since it can't be used without it